Packs/Ncurion/Integrations/Ncurion/Ncurion.py

```diff
@@ -0,0 +1,178 @@
+        verify_certificate = not demisto.params().get('insecure', False)
+        for i in log_server_id:
+            base_url_log = base_url + f'/logapi/api/v1/logserver/search/alert/search/{i}'
+            response_log = requests.request("GET", base_url_log, headers=headers1, params=params1, verify=verify_certificate)
+            data = response_log.json()
+            for hit in data:
+                incident = {
+                    'name': hit['alert']['category'] + hit['alert']['signature'],
+                    'occured': hit['@timestamp'],
+                    'rawJSON': json.dumps(hit)
+                }
+                incidents.append(incident)
+    next_run = {'last_fetch': f'{now_time}'}
+
+    logout = json.dumps({
+        "access_token": access_token,
+        "refresh_token": refresh_token
+    })
+    remove_url = base_url + '/napi/api/v1/apikey/remove'
+    requests.request("POST", remove_url, headers=headers1, data=logout, verify=verify_certificate)
+    return next_run, incidents
+
+
+def api_log_out(base_url, access_token, refresh_token, headers1):
+    logout = json.dumps({
+        "access_token": access_token,
+        "refresh_token": refresh_token
+    })
+    remove_url = base_url + '/napi/api/v1/apikey/remove'
+    verify_certificate = not demisto.params().get('insecure', False)
+    requests.request("POST", remove_url, headers=headers1, data=logout, verify=verify_certificate)
+
+
+def main():
+    params = demisto.params()
+    base_url = params.get('base_url')
+    username = params.get('username')
+    password = params.get('password')
+    
+    first_fetch_time = arg_to_datetime(
+        arg=demisto.params().get('first_fetch', '3 days'),
+        arg_name='First fetch time',
+        required=True
+    )
+    first_fetch_timestamp = int(first_fetch_time.timestamp()) if first_fetch_time else None
+    assert isinstance(first_fetch_timestamp, int)
+    
+    command = demisto.command()
+    demisto.debug(f'Command being called is {command}')
+    try:
+        if command == 'test-module':
+            api_url = base_url + '/napi/api/v1/apikey'
+            payload = json.dumps({
+                "username": username,
+                "password": password
+            })
+            headers = {
+                "Content-Type": "application/json"
+            }
+            response = requests.request("POST", api_url, data=payload, headers=headers, verify=False)
+            if response.status_code == 200:
+                return_outputs('ok')
+        elif command == 'fetch-incidents':
+            next_run, incidents = fetch_incidents(
+                base_url, username, password,
+                last_run=demisto.getLastRun(),
+                first_fetch_time = first_fetch_timestamp
+            )
+            demisto.setLastRun(next_run)
+            demisto.incidents(incidents)
+        elif command == 'ncurion-get-log-list':
+            return_outputs(get_log_list(base_url, username, password))
```

Comment:
I think it's better to add a `limit` argument if the API knows to handle it

Reply:
commands  'ncurion-get-log-list'? There are not many targets on the sensor server, so I think it's okay not to limit.